Audit fixes (#85)
morpho-org/pre-liquidation: README.md

PreLiquidation contract addresses are generated using the CREATE2 opcode, allowing for predictable address computation depending on pre-liquidation parameters.
The [`PreLiquidationAddressLib`](./src/libraries/periphery/PreLiquidationAddressLib.sol) library provides a `computePreLiquidationAddress` function, simplifying the computation of a PreLiquidation contract's address.

### Potential preLCF manipulation

> A pre-liquidation cannot pre-liquidate more than `preLCF` proportion of the position debt, however it's possible to pre-liquidate a small part of the position while keeping it pre-liquidatable before performing another pre-liquidation. This manipulation can lead to repaying more than `preLCF` of the debt. This has been studied in the part 5.2 of [An Empirical Study of DeFi Liquidations:

Reviewer: I wonder if the wording

>  more than `preLCF` proportion of the position debt

could be improved. I mean `preLCF` is a factor, so this reads out as 

>  more than [_factor_] proportion of the position debt

which I find somewhat unclear.

Author: dropped the "proportion" word, seems a bit better

> A pre-liquidation cannot pre-liquidate more than `preLCF` proportion of the position debt, however it's possible to pre-liquidate a small part of the position while keeping it pre-liquidatable before performing another pre-liquidation. This manipulation can lead to repaying more than `preLCF` of the debt. This has been studied in the part 5.2 of [An Empirical Study of DeFi Liquidations:

Reviewer: ```suggestion
A pre-liquidation cannot pre-liquidate more than `preLCF` proportion of the position debt. 
However, it's possible to pre-liquidate a fraction of the position while keeping it pre-liquidatable before performing another pre-liquidation. 
This manipulation can lead to repaying more than `preLCF` of the debt.
This has been studied in the part 5.2 of [An Empirical Study of DeFi Liquidations:
```

Reviewer: We should be careful on the one line per sentence.

> Incentives, Risks, and Instabilities](https://arxiv.org/pdf/2106.06389). Implementing a `preLCF` linear with the health factor mitigates this.

Author: I thought about it but it felt difficult to introduce the recovery close factor without writing a lot of math

> Incentives, Risks, and Instabilities](https://arxiv.org/pdf/2106.06389). Implementing a `preLCF` linear with the health factor mitigates this.

Author: Would be great if we could link an article or a paper presenting it

> Incentives, Risks, and Instabilities](https://arxiv.org/pdf/2106.06389). Implementing a `preLCF` linear with the health factor mitigates this.

Reviewer 2: i don't think that there are any paper presenting it unfortunately 

you can inspire from what's [there](https://www.notion.so/morpho-labs/On-the-Recovery-Close-Factor-104d69939e6d8017bd28d2c3ca192a35?pvs=4), it's not that complex

> Incentives, Risks, and Instabilities](https://arxiv.org/pdf/2106.06389). Implementing a `preLCF` linear with the health factor mitigates this.

Reviewer 2: btw, you can keep this for an other PR, as the readme isn't in the scope of the audits

> Incentives, Risks, and Instabilities](https://arxiv.org/pdf/2106.06389). Implementing a `preLCF` linear with the health factor mitigates this.

Reviewer 2: #86

PreLiquidation contract addresses are generated using the CREATE2 opcode, allowing for predictable address computation depending on pre-liquidation parameters.
The [`PreLiquidationAddressLib`](./src/libraries/periphery/PreLiquidationAddressLib.sol) library provides a `computePreLiquidationAddress` function, simplifying the computation of a PreLiquidation contract's address.

### Potential preLCF manipulation

> A pre-liquidation cannot pre-liquidate a proportion of the position's debt greater than `preLCF`, however it's possible to pre-liquidate a fraction of the position while keeping it pre-liquidatable before performing another pre-liquidation.

Reviewer 3: ```suggestion
A pre-liquidation cannot pre-liquidate a proportion of the position's debt greater than `preLCF`, however it's possible to pre-liquidate a proportion of the position while keeping it pre-liquidatable before performing another pre-liquidation.
```
to keep the same naming

> This manipulation can lead to repaying a proportion of the position's debt higher than `preLCF` and it has been studied in the part 5.2 of [An Empirical Study of DeFi Liquidations:Incentives, Risks, and Instabilities](https://arxiv.org/pdf/2106.06389).

Reviewer 3: ```suggestion
This manipulation can lead to repaying a proportion of the position's debt higher than `preLCF`.
It has been studied in the part 5.2 of [An Empirical Study of DeFi Liquidations:Incentives, Risks, and Instabilities](https://arxiv.org/pdf/2106.06389), in the case of a constant liquidation close factor.
```

> Implementing a `preLCF` linear with the health factor mitigates this.

Reviewer 3: ```suggestion
Implementing a `preLCF` linear in the health factor can help mitigating this manipulation when choosing the right slope.
```

PreLiquidation contract addresses are generated using the CREATE2 opcode, allowing for predictable address computation depending on pre-liquidation parameters.
The [`PreLiquidationAddressLib`](./src/libraries/periphery/PreLiquidationAddressLib.sol) library provides a `computePreLiquidationAddress` function, simplifying the computation of a PreLiquidation contract's address.

### Potential preLCF manipulation

A pre-liquidation cannot pre-liquidate more than `preLCF` of the position debt.

> However it's possible to pre-liquidate a small part of the position while keeping it pre-liquidatable before performing another pre-liquidation.

Reviewer: ```suggestion
However it's possible to pre-liquidate a fraction of the position while keeping it pre-liquidatable before performing another pre-liquidation.
```

> A pre-liquidation cannot pre-liquidate a proportion of the position's debt greater than `preLCF`, however it's possible to pre-liquidate a proportion of the position while keeping it pre-liquidatable before performing another pre-liquidation.

Reviewer: Actually, why not the following,
```suggestion
A pre-liquidation cannot repay a proportion of the position's debt greater than `preLCF`, however it's possible to pre-liquidate a proportion of the position while keeping it pre-liquidatable before performing another pre-liquidation.
```

> A pre-liquidation cannot pre-liquidate a proportion of the position's debt greater than `preLCF`, however it's possible to pre-liquidate a proportion of the position while keeping it pre-liquidatable before performing another pre-liquidation.

Reviewer: Punctuation. 
```suggestion
A pre-liquidation cannot pre-liquidate a proportion of the position's debt greater than `preLCF`.

However, it's possible to pre-liquidate a proportion of the position while keeping it pre-liquidatable before performing another pre-liquidation.
```

### Potential preLCF manipulation

A pre-liquidation cannot repay a proportion of the position's debt greater than `preLCF`.

> However it's possible to pre-liquidate a proportion of the position while keeping it pre-liquidatable before performing another pre-liquidation.

Reviewer: However,

### Potential preLCF manipulation

> A pre-liquidation cannot pre-liquidate more than `preLCF` of the position debt.

Reviewer: Sorry but this still not clear to me. What does `factor of the position debt` means?

> However it's possible to pre-liquidate a small part of the position while keeping it pre-liquidatable before performing another pre-liquidation.

Reviewer: Missing comma, see original suggestion.

> This manipulation can lead to repaying more than `preLCF` of the debt and it has been studied in the part 5.2 of [An Empirical Study of DeFi Liquidations:Incentives, Risks, and Instabilities](https://arxiv.org/pdf/2106.06389).

Reviewer: Dito, previous comment. What is `factor of the debt`.

Run `forge test`.

## Solidity version

> The main contracts (`PreLiquidation` and `PreLiquidationFactory`) are written in Solidity 0.8.27.

Reviewer: To be absolutely clear, the programs are written with Solidity and compiled using `solc` version 0.8.27. In particular, the languages includes a directive (`pragma`) to ensure a given compiler version is used. I don't know if the language specification version is the same as the compiler. Comment is optional.

Author: I think the language specification version is the same as the compiler (for instance the custom error in require is specific to Solidity 0.8.27)

Reviewer: I'm interested, do you have a link for the language specification that states the version?

Author: Not sure what you mean exactly by language specification but you can get "language features" in each solidity release version (for example [here](https://github.com/ethereum/solidity/releases/tag/v0.8.27) for 0.8.27)

Reviewer: As mater of fact, when you follow your link you'll realize that the version is attached to the compiler and not the language which are two different objects.

> The bytecode of these contracts could contain new opcodes (e.g., `PUSH0`, `MCOPY`, `TSTORE`, `TLOAD`) so one should make sure that the contract bytecode can be handled by the chain used for deployment.

Reviewer: ```suggestion
As a consequence of using version X of the compiler, the bytecode blabla
```
